src/utf.d

```diff
@@ -22,9 +22,6 @@ bool utf_isValidDchar(dchar c)
     // surrogate pairs
     if (0xD800 <= c && c <= 0xDFFF)
         return false;
-    // non-characters
```

Comment:
I'd like to see a comment here saying specifically that FFFE and FFFF are allowed for internal use, so the next maintainer won't 'fix' it back again. A link to the discussion https://issues.dlang.org/show_bug.cgi?id=1357 would be helpful as well.